src/sac/sac1.f90

```diff
@@ -0,0 +1,789 @@
+  ! ADJUST ADIMC,ADDITIONAL IMPERVIOUS AREA STORAGE, FOR EVAPORATION.
+  ADIMC = ADIMC - E5
+  
+  IF (ADIMC .LT. 0.0_dp) THEN
+    ! E5 CAN NOT EXCEED ADIMC.
+    E5 = E5 + ADIMC
+    ADIMC = 0.0_dp
+  END IF
+  ! E5 IS ET FROM THE AREA ADIMP.
+  E5 = E5 * ADIMP
+
+  ! COMPUTE PERCOLATION AND RUNOFF AMOUNTS.
+
+  ! TWX IS THE TIME INTERVAL AVAILABLE MOISTURE IN EXCESS
+  ! OF UZTW REQUIREMENTS.
+  TWX = PXV + UZTWC - UZTWM
+  
+  IF (TWX .LT. 0.0_dp) THEN
+    ! ALL MOISTURE HELD IN UZTW--NO EXCESS.
+    UZTWC = UZTWC + PXV
+    TWX = 0.0_dp
+  ELSE
+    ! MOISTURE AVAILABLE IN EXCESS OF UZTW STORAGE.
+    UZTWC = UZTWM
+  END IF
+  
+  ADIMC = ADIMC + PXV - TWX
+
+  ! COMPUTE IMPERVIOUS AREA RUNOFF.
+
+  ! ROIMP IS RUNOFF FROM THE MINIMUM IMPERVIOUS AREA.
+  ROIMP = PXV * PCTIM
+  SIMPVT = SIMPVT + ROIMP
+
+  ! INITIALIZE TIME INTERVAL SUMS.
+  SBF=0.0_dp; SSUR=0.0_dp; SIF=0.0_dp; SPERC=0.0_dp; SDRO=0.0_dp; SPBF=0.0_dp
+
+  ! DETERMINE COMPUTATIONAL TIME INCREMENTS FOR THE BASIC TIME INTERVAL
+
+  ! NINC=NUMBER OF TIME INCREMENTS THAT THE TIME INTERVAL
+  ! IS DIVIDED INTO FOR FURTHER
+  ! SOIL-MOISTURE ACCOUNTING.  NO ONE INCREMENT
+  ! WILL EXCEED 5.0 MILLIMETERS OF UZFWC+PAV  
+  NINC = INT(1.0_dp + 0.2_dp * (UZFWC + TWX))
+
+  ! DINC=LENGTH OF EACH INCREMENT IN DAYS.
+  DINC = (1.0_dp / REAL(NINC, dp)) * DT
+
+  ! PINC=AMOUNT OF AVAILABLE MOISTURE FOR EACH INCREMENT.
+  PINC = TWX / REAL(NINC, dp)
+
+  ! COMPUTE FREE WATER DEPLETION FRACTIONS FOR
+  ! THE TIME INCREMENT BEING USED-BASIC DEPLETIONS
+  ! ARE FOR ONE DAY
+  DUZ = 1.0_dp - ((1.0_dp - UZK) ** DINC)
+  DLZP = 1.0_dp - ((1.0_dp - LZPK) ** DINC)
+  DLZS = 1.0_dp - ((1.0_dp - LZSK) ** DINC)
+
+  ! INFERRED PARAMETER (ADDED BY Q DUAN ON 3/6/95)
+  PAREA = 1.0_dp - ADIMP - PCTIM
+  
+  ! -------------------------------------------------------------------
+  ! START INCREMENTAL LOOP FOR THE TIME INTERVAL
+  ! -------------------------------------------------------------------
+  DO I = 1, NINC
+    
+    ADSUR = 0.0_dp
+    ! COMPUTE DIRECT RUNOFF (FROM ADIMP AREA)
+    RATIO = (ADIMC - UZTWC) / LZTWM
+    IF (RATIO .LT. 0.0_dp) RATIO = 0.0_dp
+    ! ADDRO IS THE AMOUNT OF DIRECT RUNOFF FROM THE AREA ADIMP.
+    ADDRO = PINC * (RATIO ** 2)
+    
+    ! COMPUTE BASEFLOW AND KEEP TRACK OF TIME INTERVAL SUM.
+    BF = LZFPC * DLZP
+    LZFPC = LZFPC - BF
+    IF (LZFPC .LE. 0.0001_dp) THEN
+      BF = BF + LZFPC
+      LZFPC = 0.0_dp
+    END IF
+    SBF = SBF + BF
+    SPBF = SPBF + BF
+    
+    BF = LZFSC * DLZS
+    LZFSC = LZFSC - BF
+    IF (LZFSC .LE. 0.0001_dp) THEN
+      BF = BF + LZFSC
+      LZFSC = 0.0_dp
+    END IF
+    SBF = SBF + BF
+    
+    ! COMPUTE PERCOLATION-IF NO WATER AVAILABLE THEN SKIP
+    IF ((PINC + UZFWC) .LE. 0.01_dp) THEN
+      UZFWC = UZFWC + PINC
+      ADIMC=ADIMC+PINC-ADDRO-ADSUR
+      IF (ADIMC.GT.(UZTWM+LZTWM)) THEN
+        ADDRO=ADDRO+ADIMC-(UZTWM+LZTWM)
+        ADIMC=UZTWM+LZTWM
+      ENDIF
+      SDRO=SDRO+ADDRO*ADIMP
+      CYCLE
```

Comment:
@jduckerOWP is this the point you were indicating a change to CYCLE semantics?  See the other comments about trying to get a clear diff of the old and new code for these files, it may help evaluate the semantics better...